src/java/org/apache/cassandra/gms/ApplicationState.java

```diff
@@ -19,24 +19,25 @@
 
 public enum ApplicationState
 {
-    STATUS,
+    @Deprecated STATUS, //Deprecated and unsued in 4.0, stop publishing in 5.0, reclaim in 6.0
     LOAD,
     SCHEMA,
     DC,
     RACK,
     RELEASE_VERSION,
     REMOVAL_COORDINATOR,
-    INTERNAL_IP,
-    RPC_ADDRESS,
+    @Deprecated INTERNAL_IP, //Deprecated and unused in 4.0, stop publishing in 5.0, reclaim in 6.0
+    @Deprecated RPC_ADDRESS, // ^ Same
     X_11_PADDING, // padding specifically for 1.1
     SEVERITY,
     NET_VERSION,
     HOST_ID,
     TOKENS,
     RPC_READY,
     // pad to allow adding new states to existing cluster
-    X1,
-    X2,
+    INTERNAL_ADDRESS_AND_PORT, //Replacement for INTERNAL_IP with up to two ports
+    NATIVE_ADDRESS_AND_PORT, //Replacement for RPC_ADDRESS
```

Comment:
It's unclear to me why you renamed `RPC_ADDRESS` to `NATIVE_ADDRESS`. That make it really easy to confuse "the addr/port to be used between peers" as opposed to "the addr/port open for client apps/drivers".  Perhaps a better name is `INTERNODE_ADDRESS_AND_PORT`? (or INTERNAL_ or PEER_ or ...)

Reply:
native address is what it is now because thrift RPC is gone. Now what I remember is I did this because I thought we stopped using rpc_address in the yaml and renamed it to native_address. And now I am very confused because that doesn't seem to be the case. I was trying to consistently use native_address instead of rpc_address now that thrift is gone.

Reply:
Not sure what the optimal thing to do here is. I didn't do this out of the blue and I didn't really want to, but I felt guilty about continuing to call it RPC. It's a lot of code changes to stop using native because there are many places where I made it consistent.

Reply:
ughh, so I was completely wrong wrt the RPC_ADDRESS/NATIVE_ADDRESS thing. i misread this line:
```java
    appStates.put(ApplicationState.RPC_ADDRESS, valueFactory.rpcaddress(FBUtilities.getBroadcastRpcAddress()));
```
as this:
```java
    appStates.put(ApplicationState.RPC_ADDRESS, valueFactory.rpcaddress(FBUtilities.getBroadcastAddress()));
```

(`getBroadcastRpcAddress` vs `getBroadcastAddress`) when we do the assignment in StorageService.